framework/include/materials/DerivativeMaterialInterface.h

```diff
@@ -63,7 +64,7 @@ class DerivativeMaterialInterface : public T
    * Fetch a pointer to a material property if it exists, otherwise return null
    */
   template<typename U>
-  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name);
+  MaterialProperty<U> * getMaterialPropertyPointer(const std::string & name) __attribute__ ((deprecated));
```

Comment:
Ok, I can change this to a call to ```mooseDeprecated()```. I had used both methods for the previously deprecated methods. Maybe you should think about wrapping the attribute thingie in a moos macro. The advantage is that it applies to methods, types, and objects/variables which allows pretty fine grained deprecations.